### PR TITLE
Merge and Update Remove and Delete patterns

### PR DESCRIPTION
**Pull request by chloepoulterdesign**
Removed the Remove pattern. 😆

Combined it into the delete pattern.

Made some updates to both.

Replaced a few images to keep it inline with changes that happened in the reusable component development. 

Moved a few bits and pieces into the anatomy section as they're optional/options to choose from.

### CODE REVIEW

File: src/pages/experimental/delete-pattern/index.mdx [hunk 1]
```diff
@@ -1,94 +1,213 @@
...
-description: Deleting is a destructive action. To protect user resources, your product should support ‘undo’ where possible.
+
+title: Delete / Remove pattern
+
 ---
 
-### Deleting is a destructive action. To protect user resources, your product should support ‘undo’ where possible. If ‘undo’ support is impossible, request user confirmation before deleting a resource or asset.
+### _Deleting_ is a destructive action. To protect user resources, your product should support ‘undo’ where possible. If ‘undo’ support is impossible, request user confirmation before deleting a resource or asset.
+
+### _Removal_ is a non-destructive action. A user can remove an asset from a list or item without destroying it.
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
### _Removing_ is a non-destructive action. A user can remove an asset from a list or item without destroying it.
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 2]
```diff
@@ -1,94 +1,213 @@
...
+|        | _Low impact_    | Action is reversible or very low impact. A confirmation modal may not be required.                                        |
+| Remove | _Medium impact_ | Action can't be reversed and causes some loss. The user sees a modal and confirms the consequence of removal.             |
+|        | _Low impact_    | Action is reversible or very low impact. A confirmation modal may not be required.                                        |
 
-## High impact
+## Delete High impact
 
 A high-impact deletion cannot be reversed. The action would result in a significant loss for a user if done accidentally.
 
+When deleting is medium-impact, a confirmation dialog should be presented to the user which displays:
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
When deleting is high-impact, a confirmation dialog should be presented to the user which displays:
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 3]
```diff
@@ -1,94 +1,213 @@
...
+
+- The name of the resource
+- Consequences of the deletion
+- "This action cannot be undone."
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
- "This action cannot be undone."
- "An editable text field for the name of the resource to be entered"
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 4]
```diff
@@ -1,94 +1,213 @@
...
+
 For high-impact scenarios, a user should confirm the action by manually entering the name of the resource. The 'Delete' button is enabled when the text entered perfectly matches the resource name.
 
-The text should tell the user the consequences of the deletion and that the action cannot be undone. Optionally, when the delete button is selected it changes to `Deleting…`.
+A [Done modal](#done-modal), [Requested modal](#requested-modal) or [Notification](#notification) can be used to confirm the action.
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
Optionally a [Done modal](#done-modal), [Requested modal](#requested-modal) or [Notification](#notification) can be used to confirm the action (or a combination of one modal and a notification).
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 5]
```diff
@@ -1,94 +1,213 @@
...
-You can choose to show a notification confirming deletion is completed. This is useful when the deletion of the resource takes more than a few moments.
+## Delete Medium impact
+
+A medium-impact deletion is one that cannot be reversed, but would not be catastrophic if done accidentally. When deleting is medium-impact, a confirmation dialog should be presented to the user which displays:
+
+- The name of the resource
+- Consequences of the deletion
+- "The action cannot be undone."
+
+A [Done modal](#done-modal), [Requested modal](#requested-modal) or [Notification](#notification) can be used to confirm the action.
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
Optionally a [Done modal](#done-modal), [Requested modal](#requested-modal) or [Notification](#notification) can be used to confirm the action (or a combination of one modal and a notification).
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 6]
```diff
@@ -1,94 +1,213 @@
...
-![Example of a standard delete modal in its "deleting" state](images/1_2.png)
+<Caption> Example low impact remove options</Caption>
 
 </Column>
 </Row>
 
-You can choose to show a notification confirming deletion is completed. This is useful when the deletion of the resource takes more than a few moments.
+## Anatomy
+
+### Success modal
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
### Optional passive modal
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 7]
```diff
@@ -1,94 +1,213 @@
...
+
+In high or medium impact flows, an optional final dialog can be used to signify success at the end of a remove or delete action.
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
In high or medium impact flows, an optional passive modal can be used to signify the state at the end of a remove or delete action.
This is particularly useful when the deletion or removal doesn't happen immediately, or a request has to be sent to trigger that action. 
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 8]
```diff
@@ -1,94 +1,213 @@
...
+
+<Row>
+<Column colLg={8}>
+
+![Example of a delete successful modal](images/DoneSerious.png)
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]: I would switch this to the delete request one instead.

File: src/pages/experimental/delete-pattern/index.mdx [hunk 9]
```diff
@@ -1,94 +1,213 @@
...
+![Example of a remove successful modal](images/DoneMedium.png)
+
+<Caption> Use a success modal to show that removing is complete.</Caption>
+
+</Column>
+</Row>
+
+### Notification
+
+Use an optional notification to confirm a delete or remove action is completed. This is useful when the action takes more than a few moments.
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
An optional notification can be used to confirm a delete or remove action has completed. This is useful when the action takes more than a few moments.
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 10]
```diff
@@ -1,94 +1,213 @@
...
+  {' '}
+  An optional notification can be used to confirm the delete or remove action
+  with medium or high impact actions.
+</Caption>
+
 </Column>
 </Row>
 
-## Low impact
+### Requested modal
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]: I would remove this whole section (as my previous suggestion is to make the success modal section an optional passive modal section which would encompass this)

File: src/pages/experimental/delete-pattern/index.mdx [hunk 11]
```diff
@@ -1,94 +1,213 @@
...
+<Caption>
+  {' '}
+  An optional extra Deletion requested or Removal requested modal can be added
+  to medium or high impact workflows.
+</Caption>
+
+</Column>
+</Row>
+
+### Positioning
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]: I'd remove this whole section

File: src/pages/experimental/delete-pattern/index.mdx (hunk 8 above)
Comment (chloepoulterdesign) [reply src/pages/experimental/delete-pattern/index.mdx]: Ok, will do. Just got to figure out how to make the change!

File: src/pages/experimental/delete-pattern/index.mdx (hunk 10 above)
Comment (chloepoulterdesign) [reply src/pages/experimental/delete-pattern/index.mdx]: Will also follow this one up

File: src/pages/experimental/delete-pattern/index.mdx (hunk 11 above)
Comment (chloepoulterdesign) [reply src/pages/experimental/delete-pattern/index.mdx]: Do you mean the "positioning" section?

File: src/pages/experimental/delete-pattern/index.mdx [hunk 12]
```diff
@@ -1,94 +1,179 @@
...
-## High impact
+## Delete High impact
 
 A high-impact deletion cannot be reversed. The action would result in a significant loss for a user if done accidentally.
 
+When deleting is high-impact, a confirmation dialog should be presented to the user which displays:
+
+- The name of the resource
+- Consequences of the deletion
+- "This action cannot be undone."
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
- This action cannot be undone.
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 13]
```diff
@@ -1,94 +1,179 @@
...
+- "An editable text field for the name of the resource to be entered"
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
- An editable text field for the name of the resource to be entered
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 14]
```diff
@@ -1,94 +1,179 @@
...
 </Row>
 
-You can choose to show a notification confirming deletion is completed. This is useful when the deletion of the resource takes more than a few moments.
+## Delete Medium impact
+
+A medium-impact deletion is one that cannot be reversed, but would not be catastrophic if done accidentally. When deleting is medium-impact, a confirmation dialog should be presented to the user which displays:
+
+- The name of the resource
+- Consequences of the deletion
+- "The action cannot be undone."
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
- The action cannot be undone.
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 15]
```diff
@@ -1,94 +1,179 @@
...
+
+Optionally an [Optional passive modal](#optional-passive-modal) or [Notification](#notification) can be used to confirm the action (or a combination of one modal and a notification).
 
 <Row>
 <Column colLg={8}>
 
-![Example of a successful delete notification](images/5_2.png)
+![Example of a standard delete modal](images/6.png)
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
![Example of a medium impact delete modal](images/6.png)
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 16]
```diff
@@ -1,94 +1,179 @@
...
+<Caption>An example of a standard delete modal</Caption>
 
 </Column>
 </Row>
 
-## Medium impact
+## Delete Low impact
 
-A medium-impact deletion is one that cannot be reversed, but would not be catastrophic if done accidentally. When deleting is medium-impact, a confirmation dialog should be presented to the user which displays:
+Requiring the user to confirm deletion is generally recommended. However, in very low impact situations, such as when an ‘undo’ option is available, user confirmation may not be required.
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
Requiring the user to confirm deletion is generally recommended. However, in low impact situations, or when an ‘undo’ option is available, user confirmation may not be required.
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 17]
```diff
@@ -1,94 +1,179 @@
...
+
+![Example of delete options in context without warnings for low impact delete actions](images/LowImpactDelete.png)
+
+<Caption> Low impact delete options</Caption>
+</Column>
+</Row>
+
+## Remove Medium impact
+
+Removing has a medium impact when the action can't be reversed, but would not be catastrophic if done accidentally.
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
Removing has a medium impact when the action can't be reversed, and could have some impact if done accidentally.
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 18]
```diff
@@ -1,94 +1,179 @@
...
 </Column>
 </Row>
 
-You can choose to show a notification confirming deletion is completed. This is useful when the deletion of the resource takes more than a few moments.
+## Anatomy
+
+### Optional passive modal
+
+In high or medium impact flows, an optional passive modal can be used to signify the state at the end of a remove or delete action.
+Some deletion or removal actions don't happen immediately. When this is the case, you can use the optional requested modal.
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
Some deletion or removal actions don't happen immediately, in these situations it's recommended to inform the user through the optional passive modal.
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 19]
```diff
@@ -1,94 +1,179 @@
...
 
 <Row>
 <Column colLg={8}>
 
-![Example of a successful delete notification](images/7_updated.png)
+![Example of a delete successful modal](images/Requested.png)
+
+<Caption> Use a success modal to show that deleting is complete.</Caption>
```
Comment (vikkipaterson) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
<Caption> Use a success modal to show that a delete request has been made / is in progress.</Caption>
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 20]
```diff
@@ -1,94 +1,179 @@
 ---
-title: Delete pattern
-description: Deleting is a destructive action. To protect user resources, your product should support ‘undo’ where possible.
+
+title: Delete / Remove pattern
+
```
Comment (alisonjoseph) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
description: Deleting is a destructive action. To protect user resources, your product should support ‘undo’ where possible.
```

We need to add a description back in, this is used in the search results that shows up.

Comment (chloepoulterdesign) [reply src/pages/experimental/delete-pattern/index.mdx]: Thanks for this Alison

**Comment by chloepoulterdesign** [comment]
@alisonjoseph it looks to me like the description is in the frontmatter at the moment but not 100% sure (I'm a bit new to contributing 😂)

**Comment by chloepoulterdesign** [comment]
Hey again @alisonjoseph , finally figured out how to update the nav. 
Should all be good now.

File: src/pages/experimental/delete-pattern/index.mdx [hunk 21]
```diff
@@ -1,94 +1,179 @@
...
+| Action | Variation       | Description                                                                                                               |
+| ------ | --------------- | ------------------------------------------------------------------------------------------------------------------------- |
+| Delete | _High impact_   | Action can't be reversed and causes significant loss. The user types the resource name into the modal to confim deletion. |
+|        | _Medium impact_ | Action can't be reversed and causes some loss. The user sees a modal and confirms the consequence of deletion.            |
+|        | _Low impact_    | Action is reversible or very low impact. A confirmation modal may not be required.                                        |
+| Remove | _Medium impact_ | Action can't be reversed and causes some loss. The user sees a modal and confirms the consequence of removal.             |
+|        | _Low impact_    | Action is reversible or very low impact. A confirmation modal may not be required.                                        |
 
-## High impact
+## Delete High impact
```
Comment (connor-leech) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
## High impact deletion
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 22]
```diff
@@ -1,94 +1,179 @@
...
+<Caption>
+  {' '}
+  When the resource name is entered the Delete button is enabled.
+</Caption>
 
 </Column>
 </Row>
 
-You can choose to show a notification confirming deletion is completed. This is useful when the deletion of the resource takes more than a few moments.
+## Delete Medium impact
```
Comment (connor-leech) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
## Medium-impact deletion
```

File: src/pages/experimental/delete-pattern/index.mdx [hunk 23]
```diff
@@ -1,94 +1,179 @@
...
-![Example of a successful delete notification](images/5_2.png)
+![Example of a medium impact delete modal](images/6.png)
+
+<Caption>An example of a standard delete modal</Caption>
 
 </Column>
 </Row>
 
-## Medium impact
+## Delete Low impact
```
Comment (connor-leech) [inline src/pages/experimental/delete-pattern/index.mdx]:
```suggestion
## Low-impact deletion
```

**Comment by vikkipaterson** [comment]
Hi @connor-leech & @alisonjoseph I can't see this update on the carbon site, do you know when it will be up (or is there something we need to do?) Thanks

**Comment by alisonjoseph** [comment]
@vikkipaterson the updates get auto-deployed to the website ~15-20 min after a PR is merged, so the updates should be up there once this gets merged in.

**Comment by connor-leech** [comment]
@vikkipaterson @alisonjoseph The anchor links weren't consistent with the page headers which resulted in them breaking. I have some suggestions that are pending before I approve/merge

**Comment by chloepoulterdesign** [comment]
Hi @connor-leech I've re-phrased the "remove" subtitles to be in the same format as the deletion ones and fixed the anchor links - hopefully that'll work!

**Pull request merged**